### PR TITLE
Remove unused external dependencies

### PR DESCRIPTION
**Pull request by kushagraThapar**
* Removed unused external dependencies on commons-validator and commons-lang3

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-benchmark/pom.xml [hunk 1]
```diff
@@ -90,11 +90,6 @@ Licensed under the MIT License.
       <artifactId>log4j</artifactId>
     </dependency>
 
-    <dependency>
-      <groupId>org.apache.commons</groupId>
-      <artifactId>commons-lang3</artifactId>
-    </dependency>
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos-benchmark/pom.xml]: this is benchmark project which is an internal use only though.

**Comment by kushagraThapar** [comment]
> we have dependency on commons-lang3 from many places.
> 
> not sure how commons-lang3 can be removed without making a big modification.?
> 
> see this:
> 
> ```
> moderakh@cappuccino:~/github/azure/azure-sdk-for-java/sdk/cosmos/azure-cosmos/src/main/java$ grep lang3 -r . | grep import
> ./com/azure/cosmos/ClientSideRequestStatistics.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/Index.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/CosmosAsyncItemResponse.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/CosmosAsyncPermissionResponse.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/FeedResponse.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/DatabaseAccount.java:import org.apache.commons.lang3.ObjectUtils;
> ./com/azure/cosmos/DatabaseAccount.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/Resource.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/IncludedPath.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/ConflictResolutionMode.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/CosmosAsyncConflictResponse.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/CosmosAsyncUserResponse.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/SpatialType.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/SpatialIndex.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/CompositePath.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/RangeIndex.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/FeedResponseDiagnostics.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/ConsistencyPolicy.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/IndexingPolicy.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/CosmosPermissionProperties.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/CosmosAsyncDatabaseResponse.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/CosmosAsyncUserDefinedFunctionResponse.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/WebExceptionRetryPolicy.java:import org.apache.commons.lang3.time.StopWatch;
> ./com/azure/cosmos/implementation/RenameCollectionAwareClientRetryPolicy.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/directconnectivity/StoreClient.java:import org.apache.commons.lang3.math.NumberUtils;
> ./com/azure/cosmos/implementation/directconnectivity/GatewayAddressCache.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/directconnectivity/GatewayAddressCache.java:import org.apache.commons.lang3.tuple.Pair;
> ./com/azure/cosmos/implementation/directconnectivity/StoreReader.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/directconnectivity/StoreReader.java:import org.apache.commons.lang3.tuple.Pair;
> ./com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdReporter.java:import org.apache.commons.lang3.exception.ExceptionUtils;
> ./com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestHeaders.java:import org.apache.commons.lang3.EnumUtils;
> ./com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestHeaders.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestArgs.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/directconnectivity/HttpTransportClient.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/directconnectivity/ErrorUtils.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/directconnectivity/AddressInformation.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/directconnectivity/ResponseUtils.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/directconnectivity/ServerStoreModel.java:import org.apache.commons.lang3.EnumUtils;
> ./com/azure/cosmos/implementation/directconnectivity/HttpClientUtils.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/directconnectivity/StoreResponse.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/directconnectivity/QuorumReader.java:import org.apache.commons.lang3.tuple.Pair;
> ./com/azure/cosmos/implementation/directconnectivity/HttpUtils.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/directconnectivity/RequestHelper.java:import org.apache.commons.lang3.EnumUtils;
> ./com/azure/cosmos/implementation/directconnectivity/GoneAndRetryWithRetryPolicy.java:import org.apache.commons.lang3.time.StopWatch;
> ./com/azure/cosmos/implementation/ResourceTokenAuthorizationHelper.java:import org.apache.commons.lang3.tuple.Pair;
> ./com/azure/cosmos/implementation/RxDocumentServiceResponse.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/RxDocumentServiceRequest.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/RxDocumentClientImpl.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/Strings.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/RetryUtils.java:import org.apache.commons.lang3.time.StopWatch;
> ./com/azure/cosmos/implementation/DatabaseForTest.java:import org.apache.commons.lang3.RandomStringUtils;
> ./com/azure/cosmos/implementation/DatabaseForTest.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/QueryMetricsUtils.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/Database.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/Utils.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/AsyncDocumentClient.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/VectorSessionToken.java:import org.apache.commons.lang3.ObjectUtils;
> ./com/azure/cosmos/implementation/VectorSessionToken.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/RxGatewayStoreModel.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/Configs.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/http/HttpHeader.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/ResourceId.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/ResourceId.java:import org.apache.commons.lang3.tuple.Pair;
> ./com/azure/cosmos/implementation/TestConfigurations.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/Permission.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/Trigger.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/PathsHelper.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/PathsHelper.java:import org.apache.commons.lang3.tuple.Pair;
> ./com/azure/cosmos/implementation/ResourceResponse.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/SessionTokenHelper.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/SessionContainer.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/routing/CollectionRoutingMap.java:import org.apache.commons.lang3.tuple.ImmutablePair;
> ./com/azure/cosmos/implementation/routing/PartitionKeyRangeIdentity.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/routing/InMemoryCollectionRoutingMap.java:import org.apache.commons.lang3.tuple.ImmutablePair;
> ./com/azure/cosmos/implementation/routing/InMemoryCollectionRoutingMap.java:import org.apache.commons.lang3.tuple.Pair;
> ./com/azure/cosmos/implementation/routing/LocationHelper.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/query/metrics/ClientSideMetrics.java:import org.apache.commons.lang3.tuple.ImmutablePair;
> ./com/azure/cosmos/implementation/query/metrics/SchedulingStopwatch.java:import org.apache.commons.lang3.time.StopWatch;
> ./com/azure/cosmos/implementation/query/metrics/QueryMetricsTextWriter.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/query/metrics/QueryMetricsWriter.java:import org.apache.commons.lang3.tuple.ImmutablePair;
> ./com/azure/cosmos/implementation/query/metrics/FetchExecutionRangeAccumulator.java:import org.apache.commons.lang3.time.StopWatch;
> ./com/azure/cosmos/implementation/query/metrics/TextTable.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/query/OffsetContinuationToken.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/query/OrderByUtils.java:import org.apache.commons.lang3.tuple.Pair;
> ./com/azure/cosmos/implementation/query/ParallelDocumentQueryExecutionContext.java:import org.apache.commons.lang3.tuple.ImmutablePair;
> ./com/azure/cosmos/implementation/query/DocumentQueryExecutionContextFactory.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/query/DefaultDocumentQueryExecutionContext.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/query/DefaultDocumentQueryExecutionContext.java:import org.apache.commons.lang3.tuple.ImmutablePair;
> ./com/azure/cosmos/implementation/query/QueryInfo.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/query/Fetcher.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/query/OrderByDocumentQueryExecutionContext.java:import org.apache.commons.lang3.NotImplementedException;
> ./com/azure/cosmos/implementation/query/OrderByDocumentQueryExecutionContext.java:import org.apache.commons.lang3.tuple.ImmutablePair;
> ./com/azure/cosmos/implementation/query/DocumentProducer.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/query/DocumentProducer.java:import org.apache.commons.lang3.tuple.ImmutablePair;
> ./com/azure/cosmos/implementation/DocumentCollection.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/BaseAuthorizationTokenProvider.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/QueryMetrics.java:import org.apache.commons.lang3.tuple.ImmutablePair;
> ./com/azure/cosmos/implementation/caches/RxCollectionCache.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/implementation/caches/RxPartitionKeyRangeCache.java:import org.apache.commons.lang3.tuple.ImmutablePair;
> ./com/azure/cosmos/CosmosClientBuilder.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/HashIndex.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/CosmosAsyncTriggerResponse.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/PartitionKeyDefinition.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/CosmosAsyncStoredProcedureResponse.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/CosmosAsyncDatabase.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/CosmosClientException.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/CosmosAsyncContainerResponse.java:import org.apache.commons.lang3.StringUtils;
> ./com/azure/cosmos/CosmosTriggerProperties.java:import org.apache.commons.lang3.StringUtils;
> ```

Good point, and commons-lang3 is still coming through commons-text, and I checked the versions are same - 3.8.1 (for cosmos-lang3)

**Comment by kushagraThapar** [comment]
> we are directly using commons-lang3 so in either case if we use it as a explicit dependency or transitive dependency through some other lib either way it is used by the SDK. It is better to list explicit dependencies when they are directly used. either way not a big deal.

Yes, I am fine either way as well !

**Pull request merged**